### PR TITLE
 Serialization small updates

### PR DESCRIPTION
**Pull request by mjkkirschner**
This PR updates json serialization in dynamo to serialize and deserialize the NodeView property `IsVisible` as `ShowGeometry`, as defined in the schema.

 also starts deserializing `Replication` - looks like this was not done previously when using the node read converter, now it is done as part of node deserialization in the node read converter... I find it odd that this was not caught as part of the workspace data comparison but it was not being set on save before this change.

last, this PR sets some data on portModels for nodes from the deserialized portModels, this was not being done so graphs were losing their port guids and other list@level settings.

Tests are added for the portModel and replication properties as part of workspace comparison.

really Last! I've used this branch to generate a new `serialization.dyn` file which I will include in my other PR.

**Reviewers**:
@QilongTang @ColinDayOrg

### CODE REVIEW

File: test/DynamoCoreTests/SerializationTests.cs [hunk 1]
```diff
@@ -56,6 +56,23 @@ private void ExecutionEvents_GraphPostExecution(Session.IExecutionSession sessio
             lastExecutionDuration = (TimeSpan)session.GetParameterValue(Session.ParameterKeys.LastExecutionDuration);
         }
 
+        internal class portComparisonData
```
Comment (ColinDayOrg) [inline test/DynamoCoreTests/SerializationTests.cs]: The lowercase "p" here looks a little strange to me, is there a reason?

File: test/DynamoCoreTests/SerializationTests.cs [hunk 2]
```diff
@@ -64,31 +81,59 @@ internal class WorkspaceComparisonData
...
                     NodeTypeMap.Add(n.GUID, n.GetType());
+                    NodeReplicationMap.Add(n.GUID,n.ArgumentLacing.ToString());
 
                     var portvalues = n.OutPorts.Select(p =>
                         GetDataOfValue(n.GetValue(p.Index, controller))).ToList<object>();
 
+                    n.InPorts.ToList().ForEach(p =>
+                    {
+                        PortDataMap.Add(p.GUID,
+    new portComparisonData
```
Comment (ColinDayOrg) [inline test/DynamoCoreTests/SerializationTests.cs]: Some indentation issues here.

File: test/DynamoCoreTests/SerializationTests.cs [hunk 3]
```diff
@@ -194,6 +259,19 @@ private void CompareWorkspaces(WorkspaceComparisonData a, WorkspaceComparisonDat
                 Assert.AreEqual(countA, countB, string.Format("One {0} node has {1} outports, while the other has {2}", a.NodeTypeMap[kvp.Key], countA, countB));
             }
 
+            foreach(var portkvp in a.PortDataMap)
```
Comment (ColinDayOrg) [inline test/DynamoCoreTests/SerializationTests.cs]: You added spaces after the "foreach" for all of the "foreach" statements below, space here?

File: test/DynamoCoreTests/SerializationTests.cs [hunk 4]
```diff
@@ -143,6 +188,26 @@ private void CompareWorkspacesDifferentGuids(WorkspaceComparisonData a, Workspac
...
+                var newGuid = GuidUtility.Create(GuidUtility.UrlNamespace, this.modelsGuidToIdMap[portkvp.Key]);
+                Assert.IsTrue(b.PortDataMap.ContainsKey(newGuid));
+                var aPort = a.PortDataMap[portkvp.Key];
+                var bPort= b.PortDataMap[newGuid];
+                Assert.AreEqual(aPort.UseLevels,bPort.UseLevels);
+                Assert.AreEqual(aPort.KeepListStructure, bPort.KeepListStructure);
+                Assert.AreEqual(aPort.Level, bPort.Level);
+            }
+
+            foreach(var kvp in a.NodeReplicationMap)
```
Comment (ColinDayOrg) [inline test/DynamoCoreTests/SerializationTests.cs]: You added spaces after the "foreach" for all (most) of the other "foreach" statements, space here?

**Comment by ColinDayOrg** [comment]
Other than a couple of minor comments and the AppVeyor issue, it's looking good.

File: test/DynamoCoreTests/SerializationTests.cs [hunk 5]
```diff
@@ -64,31 +81,59 @@ internal class WorkspaceComparisonData
...
+                    n.InPorts.ToList().ForEach(p =>
+                    {
+                        PortDataMap.Add(p.GUID,
+    new portComparisonData
+    {
+        ID = p.GUID.ToString(),
+        UseLevels = p.UseLevels,
+        KeepListStructure = p.KeepListStructure,
+        Level = p.Level
+    });
```
Comment (QilongTang) [inline test/DynamoCoreTests/SerializationTests.cs]: Format?

**Comment by mjkkirschner** [comment]
@QilongTang @ColinDayOrg addressed the comments can we merge?

**Comment by ColinDayOrg** [comment]
LGTM

**Pull request merged**